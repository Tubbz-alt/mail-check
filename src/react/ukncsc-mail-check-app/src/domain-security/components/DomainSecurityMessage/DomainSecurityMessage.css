.DomainSecurityMessage--container {
  border-width: 0px 0px 0px 8px;
  border-style: solid;
  max-width: 600px; 
}
