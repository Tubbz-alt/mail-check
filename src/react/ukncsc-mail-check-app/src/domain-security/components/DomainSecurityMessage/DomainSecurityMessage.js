import React from 'react';
import MarkDown from 'react-markdown';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';
import { removeMailtoHyperlinksRenderer as link } from 'common/helpers/markdown';

import './DomainSecurityMessage.css';

const DomainSecurityMessage = ({ children, markdown, ...props }) => (
  <Message
    {...props}
    style={{ margin: '0.2em 0em' }}
    className="DomainSecurityMessage--container"
  >
    {markdown ? <MarkDown source={children} renderers={{ link }} /> : children}
  </Message>
);

DomainSecurityMessage.defaultProps = {
  error: false,
  warning: false,
  success: false,
  info: false,
  markdown: false,
  size: null,
};

DomainSecurityMessage.propTypes = {
  error: PropTypes.bool,
  warning: PropTypes.bool,
  success: PropTypes.bool,
  info: PropTypes.bool,
  markdown: PropTypes.bool,
  size: PropTypes.string,
};

export default DomainSecurityMessage;
