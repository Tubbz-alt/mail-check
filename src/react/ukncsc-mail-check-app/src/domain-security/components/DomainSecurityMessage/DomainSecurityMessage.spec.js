import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';

import DomainSecurityMessage from './DomainSecurityMessage';

describe('DomainSecurityMessage', () => {
  let container;

  describe('when provided with no props', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityMessage>
          <p>security rulez</p>
        </DomainSecurityMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });

  describe('when provided with the markdown flag', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityMessage markdown>
          {`- list item\n- another list item`}
        </DomainSecurityMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('render the markdown as html', () =>
      expect(container.getElementsByTagName('li')).toHaveLength(2));
  });

  describe('when provided with the markdown flag and a mailto address', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityMessage markdown>
          mailto:me@ncsc.gov.uk
        </DomainSecurityMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not render a hyperlink', () =>
      expect(container.getElementsByTagName('a')).toHaveLength(0));

    test('it should contain the full mailto text', () =>
      expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
        'mailto:me@ncsc.gov.uk'
      ));
  });

  describe('when provided with the markdown flag and a link', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityMessage markdown>
          https://ncsc.gov.uk
        </DomainSecurityMessage>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should render a hyperlink', () =>
      expect(container.getElementsByTagName('a')[0]).toHaveTextContent(
        'https://ncsc.gov.uk'
      ));
  });
});
