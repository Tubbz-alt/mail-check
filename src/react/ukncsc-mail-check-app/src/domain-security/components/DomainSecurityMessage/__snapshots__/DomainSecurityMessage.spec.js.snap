// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityMessage when provided with no props it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em;"
    class="ui message DomainSecurityMessage--container"
  >
    <p>
      security rulez
    </p>
  </div>
</div>
`;

exports[`DomainSecurityMessage when provided with the markdown flag and a link it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em;"
    class="ui message DomainSecurityMessage--container"
  >
    <div>
      <p>
        <a
          href="https://ncsc.gov.uk"
        >
          https://ncsc.gov.uk
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`DomainSecurityMessage when provided with the markdown flag and a mailto address it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em;"
    class="ui message DomainSecurityMessage--container"
  >
    <div>
      <p>
        mailto:me@ncsc.gov.uk
      </p>
    </div>
  </div>
</div>
`;

exports[`DomainSecurityMessage when provided with the markdown flag it should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em;"
    class="ui message DomainSecurityMessage--container"
  >
    <div>
      <ul>
        <li>
          list item
        </li>
        <li>
          another list item
        </li>
      </ul>
    </div>
  </div>
</div>
`;
